xds/src/main/java/io/grpc/xds/XdsServerWrapper.java

```diff
@@ -113,6 +113,14 @@ public void uncaughtException(Thread t, Throwable e) {
   private DiscoveryState discoveryState;
   private volatile Server delegate;
 
+  // Must be updated in the sync context.
```

Comment:
It isn't a thread-safe map, so all accesses should be in the sync context?

Reply:
ah, right!